aws/product/awsec2product.go

```diff
@@ -0,0 +1,191 @@
+	dbAwsRegion, err := models.AwsRegionByPretty(db.Db, value.(string))
+	if err != nil {
+		var newDbAwsRegion models.AwsRegion
+		logger.Error("Error during dbAwsRegion fetching", err)
+		newDbAwsRegion.Pretty = value.(string)
+		newDbAwsRegion.Region = value.(string)
+		if err := newDbAwsRegion.Insert(db.Db); err != nil {
+			logger.Error("Error during dbAwsRegion inserting", err)
+		}
+		dbAwsRegion = &newDbAwsRegion
+	}
+	dbAwsProduct.RegionID = dbAwsRegion.ID
+}
+
+// storeAttributeGenericInt stores an int value in the struct
+// which will be imported in the database.
+func storeAttributeGenericInt(name string, value interface{}, dbAwsProduct *models.AwsProductEc2, logger jsonlog.Logger) {
+	if value, ok := value.(string); ok {
+		if value, ok := strconv.Atoi(value); ok == nil {
+			switch name {
+			case "vcpu":
+				dbAwsProduct.Vcpu = value
+			case "ecu":
+				dbAwsProduct.Ecu = value
+			}
+		}
+	}
+}
+
+// storeAttributeGenericBool stores an bool value in the struct
+// which will be imported in the database.
+func storeAttributeGenericBool(name string, value interface{}, dbAwsProduct *models.AwsProductEc2, logger jsonlog.Logger) {
+	if value, ok := value.(string); ok {
+		var val int
+		if "Yes" == value {
+			val = 1
+		}
+		switch name {
+		case "currentGeneration":
+			dbAwsProduct.CurrentGeneration = val
+		}
+	}
+}
+
+// storeAttributeGenericString stores an string value in the struct
+// which will be imported in the database.
+func storeAttributeGenericString(name string, value interface{}, dbAwsProduct *models.AwsProductEc2, logger jsonlog.Logger) {
+	if value, ok := value.(string); ok {
+		switch name {
+		case "instanceType":
+			dbAwsProduct.InstanceType = value
+		case "memory":
+			dbAwsProduct.Memory = value
+		case "storage":
+			dbAwsProduct.Storage = value
+		case "networkPerformance":
+			dbAwsProduct.NetworkPerformance = value
+		case "tenancy":
+			dbAwsProduct.Tenancy = value
+		case "operatingSystem":
+			dbAwsProduct.OperatingSystem = value
+		}
+	}
+}
+
+// importResult parses the body returned by downloadJSON and
+// inserts the pricing in the database.
+func importResult(reader io.ReadCloser, logger jsonlog.Logger) {
+	defer reader.Close()
+	body, err := ioutil.ReadAll(reader)
+	if err != nil {
+		logger.Error("Body reading failed", err)
+	}
+	var bodyMap map[string]interface{}
+	if err = json.Unmarshal(body, &bodyMap); err != nil {
+		logger.Error("JSON Unmarshal failed", err)
+		return
+	}
+	for _, product := range bodyMap["products"].(map[string]interface{}) {
+		var dbAwsProduct models.AwsProductEc2
+		product := product.(map[string]interface{})
+		dbAwsProduct.Sku = product["sku"].(string)
+		attributes := product["attributes"].(map[string]interface{})
+		for name, attribute := range attributes {
+			if fct, ok := storeAttribute[name]; ok {
+				fct(name, attribute, &dbAwsProduct, logger)
+			}
+		}
+		dbAwsProduct.Insert(db.Db)
```

Comment:
As stated in the function prototype : `func (ape *AwsProductEc2) Insert(db XODB) error {` this method can return an error but you do not check for it
This also means that this functions must return an error